### DISCUSSION TITLE
How to use for MQTT sensor?

### DISCUSSION THREAD

**Discussion by fgonza2** [andrew-codechimp/HA-Battery-Notes#1597] [General]
I have some MQTT leak sensors.  They are multiple entities, not a unified 'device' can i add a battery note to it? so i can track the last replaced date? for example the sensor has two entities, a binary sensor that says if the leak was detected or not and another sensor that sends the percentage of the battery.   They come as two separate standalone entities.   I have many examples of thos as they are template MQTT sensors coming from RTL433.   They do NOT appear in the list of 'device' when trying to add manually as they are not a device.   So the question is how to use batttery notes with an entity, as there are many that are not devices.

**Comment by andrew-codechimp** [comment]
Hi, Battery Notes requires a device, it's how I designed it to allow the library to work.  It's not possible to add battery notes to a stand alone entity.
I keep thinking about this, but I don't have an approach yet.

**Comment by fgonza2** [comment]
What about adding an attribute to an existing entity On May 2, 2024, at 00:37, Andrew Jackson ***@***.***> wrote:﻿
Hi, Battery Notes requires a device, it's how I designed it to allow the library to work.  It's not possible to add battery notes to a stand alone entity.
I keep thinking about this, but I don't have an approach yet.

**Reply by andrew-codechimp** [reply]
No you can't add an attribute to an existing entity from another integration unfortunately.
If I can do it, it will replicate the existing device type ones with additional entities, just creating entities without a device.

**Comment by fgonza2** [comment]
What if you create a new “device” that includes the original entity sensor plus all the battery-notes entities that together form a new device.  So user will still be able to use the original entity w no changes and a new device is created that includes a mirror of the original entity plus all the BN stuff that way you still can associate it.I think this is a major feature because there are many entities that are not devices so the usage of BN is unfortunately only limited to devices.   Then there is also the case to be made to entities that do not report a battery state but you could use a manual timer o a number of days for you to estimate the battery and proactively change it after a predetermined amount of time. On May 5, 2024, at 01:54, Andrew Jackson ***@***.***> wrote:﻿
No you can't add an attribute to an existing entity from another integration unfortunately.
If I can do it, it will replicate the existing device type ones with additional entities, just creating entities without a device.

**Comment by andrew-codechimp** [comment]
Creating a device for another integration is problematic, I've tried that.
I had a breakthrough yesterday and have a way forward, but no promises when that will be ready.

**Comment by andrew-codechimp** [comment]
I have just released version 2.3 into beta which now supports entities as well as devices.
You will have to turn on betas within HACS to get it.

Ensure you read the release note for potential breaking changes with automations
https://github.com/andrew-codechimp/HA-Battery-Notes/releases/tag/2.3.0
